src/js/edu-benefits/utils/validations.js

```diff
@@ -245,11 +245,19 @@ function isValidBenefitsInformationPage(data) {
   return data.chapter33 || data.chapter30 || data.chapter32 || data.chapter1606;
 }
 
+function isValidRelinquishedDate(field) {
+  // Allow dates up to two years ago
+  const pastDate = moment().subtract(2, 'years');
+  const date = dateToMoment(field);
+
+  return !isBlankDateField(field) && date.isAfter(pastDate);
```

Comment:
This is kind of a weird once since we prefill today's date here, so it's not blank unless the user deletes it.

Good call on .isValid().  I'll add that as well.